### PR TITLE
#1272 Public View Page to be seen for not signed users

### PR DESCRIPTION
**Pull request by desfero**

### CODE REVIEW

File: app/components/App.tsx [hunk 1]
```diff
@@ -37,13 +35,10 @@ class AppComponent extends React.Component<IStateProps> {
...
     return (
       <>
-        <Header />
-        <div className="wrapper">
-          <ScrollToTop>
-            <AppRouter />
-          </ScrollToTop>
-        </div>
-        <Footer />
+        <ScrollToTop>
```
Comment (desfero) [inline app/components/App.tsx]: @Moejoe90 @lizonp there are some changes to the layout to get rid of Header and Footer from ETO preview. Therefore if you can please review them as those are global app changes and maybe I missed something as it's my second day in the project. Thanks

File: app/components/eto/registration/pages/EtoTerms.tsx [hunk 2]
```diff
@@ -33,7 +32,16 @@ interface IDispatchProps {
 
 type IProps = IStateProps & IDispatchProps;
 
-const currencies = ["eur_t", "eth"];
+interface ICurrencies {
+  [key: string]: string;
+}
+
+const CURRENCIES: ICurrencies = {
```
Comment (desfero) [inline app/components/eto/registration/pages/EtoTerms.tsx]: @trival @Moejoe90 I think it's worth to move 'CURRENCY' to some shared places as it's not specific to that component. Do you know some place where we can move that?

File: app/components/eto/EtoPreview.tsx [hunk 3]
```diff
@@ -1,35 +1,35 @@
...
+}
+
 interface IDispatchProps {
   loadEtoPreview: (previewCode: string) => void;
 }
 
-type IProps = IStateProps & IDispatchProps;
+type IProps = IStateProps & IDispatchProps & IRouterParams;
 
 class EtoPreviewComponent extends React.Component<IProps> {
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/eto/EtoPreview.tsx]: From my side I like Recompose I see that it takes logic out of the component and makes code cleaner. 

What I would suggest is we keep this as is now to not break consistency and introduce this change later on with bigger refactoring for other components. In addition to adding it to our platform development guidelines. 

Or if you think this wouldn't take much time then we can introduce it now with this component as an example on how to use it

[the remark this answers is not in the document]
Comment (desfero) [reply app/components/eto/EtoPreview.tsx]: @Moejoe90 great, sound like a plan, thanks

File: package.json [hunk 4]
```diff
@@ -19,7 +19,6 @@
     "@cypress/webpack-preprocessor": "2.0.1",
     "@storybook/addon-actions": "^3.4.10",
     "@storybook/react": "^4.0.0-alpha.16",
-    "@types/bignumber.js": "^5.0.0",
```
[the remark this answers is not in the document]
Comment (desfero) [reply package.json]: yes, types for bignumber are provided by the package https://github.com/MikeMcl/bignumber.js/blob/master/package.json#L26

**Comment by desfero** [comment]
@SergejSoft noooo, again conflict :D

File: app/utils/withContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import * as React from "react";
+
+export const withContainer: (
```
Comment (Moejoe90) [inline app/utils/withContainer.tsx]: :+1:

File: app/lib/api/eto/EtoApi.ts [hunk 6]
```diff
@@ -15,32 +15,35 @@ const ETO_PREVIEW_PATH = "/eto-previews/";
 
 @injectable()
 export class EtoApi {
-  constructor(@inject(symbols.authorizedJsonHttpClient) private httpClient: IHttpClient) {}
+  constructor(
+    @inject(symbols.authorizedJsonHttpClient) private authorizedHttpClient: IHttpClient,
+    @inject(symbols.jsonHttpClient) private httpClient: IHttpClient,
```
Comment (Moejoe90) [inline app/lib/api/eto/EtoApi.ts]: :+1:

File: app/components/eto/registration/pages/EtoTerms.tsx (hunk 2 above)
Comment (Moejoe90) [reply app/components/eto/registration/pages/EtoTerms.tsx]: Let's discuss this tomorrow

**Pull request merged**